source/codegen/templates/service_helpers.mako

```diff
@@ -85,6 +85,68 @@ ${set_response_values(output_parameters)}\
       return ::grpc::Status::OK;\
 </%def>
 
+
+<%def name="define_async_callback_method_body(function_name, function_data, parameters, config)">\
+<%
+  (input_parameters, callback_parameters) = service_helpers.get_callback_method_parameters(function_data)
+  response_parameters = common_helpers.filter_parameters_for_grpc_fields(callback_parameters)
+  request_type = service_helpers.get_request_type(function_name)
+  response_type = service_helpers.get_response_type(function_name)
+  driver_library_interface = common_helpers.get_library_interface_type_name(config)
+%>\
+    using CallbackRouter = nidevice_grpc::CallbackRouter<int32, ${service_helpers.create_param_type_list(callback_parameters)}>;
+    class ${function_name}Reactor : public nidevice_grpc::ServerWriterReactor<${response_type}, nidevice_grpc::CallbackRegistration> {
+    public:
+    ${function_name}Reactor(const ${request_type}& request, ${driver_library_interface}* library, const ResourceRepositorySharedPtr& session_repository)
+    {
+      auto status = start(&request, library, session_repository);
+      if (!status.ok()) {
```

Comment:
Note: the control flow of start has to change to reuse the enum initialization methods that include an early return. So now, rather than returning the `unique_ptr` to the producer and conditionally calling `Finish` internally; `start` will call `set_producer` internally (on success) and return a status.